webviz_subsurface/ert_jobs/smry2arrow.py

```diff
@@ -0,0 +1,170 @@
+    eclsum: EclSum, column_names: Iterable[str]
+) -> Dict[str, dict]:
+    """Builds dictionary containing metadata for all the specified summary columns"""
+    smry_meta = {}
+
+    for col_name in column_names:
+        col_meta = {}
+        col_meta["unit"] = eclsum.unit(col_name)
+        col_meta["is_total"] = eclsum.is_total(col_name)
+        col_meta["is_rate"] = eclsum.is_rate(col_name)
+        col_meta["is_historical"] = eclsum.smspec_node(col_name).is_historical()
+        col_meta["keyword"] = eclsum.smspec_node(col_name).keyword
+        col_meta["wgname"] = eclsum.smspec_node(col_name).wgname
+
+        num = eclsum.smspec_node(col_name).get_num()
+        if num is not None:
+            col_meta["get_num"] = num
+
+        smry_meta[col_name] = col_meta
+
+    return smry_meta
+
+
+def _load_smry_into_table(smry_filename: str) -> pa.Table:
+    """
+    Reads data from SMRY file into PyArrow Table.
+    DATE column is stored as an Arrow timetamp with ms resolution, timestamp[ms]
+    All numeric columns will be stored as 32 bit float
+    Summary meta data will be attached per field/column of the table's schema under the
+    'smry_meta' key
+    """
+
+    eclsum = EclSum(smry_filename, include_restart=False, lazy_load=False)
+
+    # Unclear what the difference between these two is, but it seems that
+    # EclSum.pandas_frame() internally uses EclSumKeyWordVector
+    # For now, we go via a set to prune out duplicate entries being returned by EclSumKeyWordVector,
+    # see: https://github.com/equinor/ecl/issues/816#issuecomment-865881283
+    column_names: List[str] = list(set(EclSumKeyWordVector(eclsum, add_keywords=True)))
+    # column_names = eclsum.keys()
+
+    # Exclude CPI columns from export
+    org_col_count = len(column_names)
+    column_names = [colname for colname in column_names if not _is_cpi_column(colname)]
+    if len(column_names) != org_col_count:
+        logger.info(
+            f"Excluding {org_col_count - len(column_names)} CPI columns from export"
+        )
+
+    # Fetch the dates as a numpy array with ms resolution
+    np_dates_ms = eclsum.numpy_dates
+
+    smry_meta_dict = _create_smry_meta_dict(eclsum, column_names)
+
+    # Datatypes to use for DATE column and all the numeric columns
+    dt_timestamp_ms = pa.timestamp("ms")
+    dt_float32 = pa.float32()
+
+    # Build schema for the table
+    field_list: List[pa.Field] = []
+    field_list.append(pa.field("DATE", dt_timestamp_ms))
+    for colname in column_names:
+        field_metadata = {b"smry_meta": json.dumps(smry_meta_dict[colname])}
+        field_list.append(pa.field(colname, dt_float32, metadata=field_metadata))
+
+    schema = pa.schema(field_list)
+
+    # Now extract all the summary vectors one by one
+    # We do this through EclSum.numpy_vector() instead of EclSum.pandas_frame() since
+    # the latter throws an exception if the SMRY data has timestamps beyond 2262,
+    # see: https://github.com/equinor/ecl/issues/802
```

Comment:
```suggestion
    # Now extract all the summary vectors one by one
    # We do this through EclSum.numpy_vector() instead of EclSum.pandas_frame() since
    # the latter throws an exception if the SMRY data has timestamps beyond 2262,
    # see: https://github.com/equinor/ecl/issues/802
```
Regardless of 2262, was the code also shorter with the numpy approach? I.e. preferable anyway?

Reply:
The code is shorter **and** cleaner when using `EclSum.numpy_vector()`, but Ecl documentation warns that for the retrieval of many vectors,  `EclSum.pandas_frame()` will be more performant.